src/intl-head.tsx

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from 'next/router';
+import React from 'react';
+import Head from 'next/head';
+import { ReactNode, ReactElement } from 'react';
+import { useGetOriginUrl } from './hooks/useGetOriginUrl';
+import { useGetAlternateLinks } from './hooks/useGetAlternateLinks';
+import { useGetCanonicalUrl } from './hooks/useGetCanonicalUrl';
+
+interface IntlHeadProps {
+  children: ReactNode;
+  title: string;
+  language?: string;
+}
+
+export function IntlHead({ children, title, language }: IntlHeadProps): ReactElement {
+  const { locale, query, asPath, locales, pathname } = useRouter();
+
+  const canonicalUrl = useGetCanonicalUrl({ locale: language ?? locale, asPath, query });
+  const alternateLinks = useGetAlternateLinks({
+    locales,
+    locale: language ?? locale,
+    asPath,
+    pathname,
+  });
+  const origin = useGetOriginUrl();
+
+  return (
+    <Head>
+      <title>{title}</title>
+      {alternateLinks}
+      {pathname !== '/' && <link rel="canonical" href={`${origin}${canonicalUrl}`} />}
+      {children}
+    </Head>
+  );
+}
```

Comment:
I'd refactor this into something like `AlternateLinks`, an element that the user then places inside their own `Head`. It's also a bit risky to hard-code in assumptions like special-casing the `/` path about the site always being served from the domain root.

Reply:
Hi @eemeli,

Indeed, I've assumed that sites would be served at '/'

How would you go about defining what's the root of the site? Would you have a prop `root` and if `true` you'd then have
```
<link rel="alternate" href="https://my-site.com/fr-CA/" hreflang="fr-CA">
<link rel="alternate" href="https://my-site.com" hreflang="x-default">
<link rel="alternate" href="https://my-site.com/en-CA" hreflang="en-CA">
```

Reply:
It should be available as `basePath` in the router object.

Reply:
Hi @eemeli,

I've added the ability to handle `basePath`. Now the reason I haven't refactored IntlHead for AlternateLinks it's because I'd like to have a single solution that would take care of the alternate links and alternate canonical. 

I'm going to check with @nbouvrette and see what he thinks about it.

I do understand your point though.